### PR TITLE
Detect whether TPOT is running in a notebook

### PR DESCRIPTION
**Pull request by weixuanfu**
Related issue #723. Credits to @rhiever

### CODE REVIEW

File: tpot/base.py [hunk 1]
```diff
@@ -94,6 +93,25 @@ def handler(dwCtrlType, hook_sigint=_thread.interrupt_main):
 
     win32api.SetConsoleCtrlHandler(handler, 1)
 
+def isnotebook():
```
Comment (rhiever) [inline tpot/base.py]: Suggestion: Refactor to `is_notebook()`.

File: tpot/base.py [hunk 2]
```diff
@@ -94,6 +93,25 @@ def handler(dwCtrlType, hook_sigint=_thread.interrupt_main):
...
+    """Check if TPOT is running in Jupyter notebook."""
```
Comment (rhiever) [inline tpot/base.py]: Suggestion: Add docstrings to clarify what this function is returning.

File: tpot/base.py (hunk 1 above)
Comment (weixuanfu) [reply tpot/base.py]: OK

File: tpot/base.py [hunk 3]
```diff
@@ -94,6 +93,29 @@ def handler(dwCtrlType, hook_sigint=_thread.interrupt_main):
...
+    -------
+    True: TPOT is running in Jupyter notebook
+    False: TPOT is running in Default Terminal/IPython Terminal/Other type
+    """
+    try:
+        from IPython import get_ipython
+        shell = get_ipython().__class__.__name__
+        # if shell == 'TerminalInteractiveShell', then Terminal running IPython
+        if shell == 'ZMQInteractiveShell':
+            return True   # Jupyter notebook or qtconsole
```
Comment (rhiever) [inline tpot/base.py]: Seems these if statements could be collapsed to

```python
return shell == 'ZMQInteractiveShell'
```

**Comment by rhiever** [comment]
LGTM

**Pull request merged**